pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/pubsub/WorkflowStatusUpdate.java

```diff
@@ -88,46 +90,56 @@ public static void updateCustomWorkflow(Map<String, String> attributesMap, Strin
+                                             DDPInstance instance) {
+        String instanceName = instance.getName();
+        List<ParticipantData> participantDataList =
+                participantDataDao.getParticipantDataByParticipantId(ddpParticipantId);
         Optional<FieldSettingsDto> fieldSetting =
-                fieldSettingsDao.getFieldSettingByColumnNameAndInstanceId(Integer.parseInt(instance.getDdpInstanceId()), workflow);
+                fieldSettingsDao.getFieldSettingByColumnNameAndInstanceId(instance.getDdpInstanceIdAsInt(), workflow);
         if (fieldSetting.isEmpty()) {
-            throw new DsmInternalError(String.format("Invalid workflow name %s for instance %s", workflow, instance.getName()));
+            throw new DsmInternalError(String.format("Invalid workflow name %s for instance %s", workflow, instanceName));
         }
 
         FieldSettingsDto setting = fieldSetting.get();
         String fieldType = setting.getFieldType();
-        boolean isOldParticipant = participantDatas.stream().anyMatch(
+        boolean isOldParticipant = participantDataList.stream().anyMatch(
                 participantDataDto -> {
                     if (participantDataDto.getFieldTypeId().isPresent()) {
                         String ft = participantDataDto.getFieldTypeId().get();
                         return ft.equals(fieldType) || ft.contains(FamilyMemberConstants.PARTICIPANTS);
                     }
                     return false;
                 });
+
         if (isOldParticipant) {
-            participantDatas.forEach(participantDataDto -> {
-                log.info("Updating participant {} data for participant {} in study {} via workflow {} with status {}",
-                        fieldType, ddpParticipantId, studyGuid, workflow, status);
-                updateProbandStatus(workflow, status, participantDataDto, fieldType);
+            participantDataList.forEach(participantData -> {
+                log.info("Updating participantData {} for participant {} in study {} via workflow {} with status {}",
+                        fieldType, ddpParticipantId, instanceName, workflow, status);
+                updateProbandStatus(workflow, status, participantData, fieldType);
             });
         } else {
-            log.info("Creating participant {} data for participant {} in study {} via workflow {} with status {}",
-                    fieldType, ddpParticipantId, studyGuid, workflow, status);
+            log.info("Creating participantData {} for participant {} in study {} via workflow {} with status {}",
+                    fieldType, ddpParticipantId, instanceName, workflow, status);
             createParticipantData(workflow, status, ddpParticipantId, instance.getDdpInstanceIdAsInt(), fieldType);
         }
-        exportWorkflowToES(workflow, status, ddpParticipantId, instance, setting, participantDatas);
 
-        try {
-            if (isATStudy(studyGuid)) {
```

Comment:
This is bug #2 that masked bug #1, correct?

Reply:
Also I think we both are learning something about markdown pound-number issue link macros today 🙄

Reply:
This is actually just an unfortunate git diff. Mostly what I did here was break out a hunk of code in `updateWorkflowData` to `updateWorkflowData`. Overdue since it helps readability and testing. (As I mentioned, I've been refactoring this spaghetti code for a while, but been conservative until I can pair changes with tests.)
I did not intend to use #1 and #2 as markdown. The two bugs have nothing to do with each other.